gunrock/app/color/color_test.cuh

```diff
@@ -68,36 +69,69 @@ auto seed     = parameters.Get<int>("seed");
-            VertexT min = v;
-            float temp  = rand[v];
-            
-            for (SizeT e = start_edge; e < start_edge + num_neighbors; e++) {
-                VertexT u = graph.GetEdgeDest(e);
-                if (rand[u] > temp)
-                    max = u;
-                
-                if (rand[u] < temp)
-                    min = u;
-                
-                printf("Let's see what rand[u] = %f\n", rand[u]);
-                temp = rand[u];
-            }
-            
-            if (colors[max] == -1)
-                colors[max] = iteration*2+1;
-
-            if (colors[min] == -1)
-                colors[min] = iteration*2+2;
-
-            printf("iteration number = %u\n", iteration);
-            printf("colors[%u, %u] = [%u, %u]\n", min, max, colors[min], colors[max]);
+          if(colors[v] != -1)
+            continue;
+          SizeT start_edge 	= graph.GetNeighborListOffset(v);
+          SizeT num_neighbors = graph.GetNeighborListLength(v);
+          float temp  = rand[v];
+          bool colormax = true;
+          bool colormin = true;
+          for (SizeT e = start_edge; e < start_edge + num_neighbors; e++) {
+              VertexT u = graph.GetEdgeDest(e);
+              if ((colors[u] == -1) && (rand[u] >= temp)) {
+                  printf("Max: Node %d with %f defeated by node %d with %f\n",
+                  v,rand[v],u,rand[u]);
+                  colormax = false;
+              }
+
+              if ((colors[u] == -1) && (rand[u] <= temp)) {
+                  printf("Min: Node %d with %f defeated by node %d with %f\n",
+                  v,rand[v],u,rand[u]);
+                  colormin = false;
+              }
+          }
+
+          if(colormax)
+            colors[v] = iteration*2+1;
+          if(colormin)
+            colors[v] = iteration*2+2;
         }
+      }
+    } else {
+      for (int iteration = 0; iteration < usr_iter; iteration++) {
+          for (VertexT v = 0; v < graph.nodes; v++) {
+              SizeT start_edge 	= graph.GetNeighborListOffset(v);
+              SizeT num_neighbors = graph.GetNeighborListLength(v);
+
+              VertexT max = v;
+              VertexT min = v;
+              float temp  = rand[v];
+
+              for (SizeT e = start_edge; e < start_edge + num_neighbors; e++) {
+                  VertexT u = graph.GetEdgeDest(e);
+                  if (rand[u] > temp)
+                      max = u;
+
+                  if (rand[u] < temp)
+                      min = u;
+
+                  printf("Let's see what rand[u] = %f\n", rand[u]);
+                  temp = rand[u];
+              }
+
+              if (colors[max] == -1)
+                  colors[max] = iteration*2+1;
+
+              if (colors[min] == -1)
+                  colors[min] = iteration*2+2;
+
+              printf("iteration number = %u\n", iteration);
+              printf("colors[%u, %u] = [%u, %u]\n", min, max, colors[min], colors[max]);
```

Comment:
Make sure there are no printfs in the final version, or they are under a DEBUG flag or something.